relay-server/src/actors/store.rs

```diff
@@ -953,6 +959,8 @@ struct MetricKafkaMessage {
     org_id: u64,
     project_id: ProjectId,
     name: String,
+    #[serde(skip)]
+    headers: BTreeMap<String, String>,
```

Comment:
nit: I would find it slightly cleaner to make the `KafkaMessage` type a struct with fields `headers` and a `payload` enum, but up to you.

Reply:
I like the idea to extract headers from the message, but also would like each type of `KafkaMessage` have its own set of headers. Refactored it a little bit to make the intention a little bit clearer.